subsys/usb/usb_c/Kconfig

```diff
@@ -39,6 +39,13 @@ config BUILD_OUTPUT_VIF
 	help
 	  Generate XML file containing VIF policies during project build.
 
+config GENVIF_INPUT_VIF_XML_PATH
+	string "Path to input VIF file"
+	depends on BUILD_OUTPUT_VIF
+	help
+	  Path of the GenVIF input VIF XML file relative to the application source
+	  directory.
+
```

Comment:
GENVIF_INPUT_VIF_XML_PATH holds path to a single xml file like "subboard/src/sink_vif.xml". At present it doesn't support multiple files. The single file path configured in prj.conf will be looked up in application directory.

This config options comes handy when project structure of users is complex. I mean storing vif source file in subdirectories with different names.

Reply:
> GENVIF_INPUT_VIF_XML_PATH holds path to a single xml file like "subboard/src/sink_vif.xml". At present it doesn't support multiple files. The single file path configured in prj.conf will be looked up in application directory.

not sure the question was fully understood.
So let me try to rephrase.

Use case 1:
- prj.conf: GENVIF_INPUT_VIF_XML_PATH=vif.xml
- prj_alt1.conf: GENVIF_INPUT_VIF_XML_PATH=vif_alt1.xml
- prj_alt2.conf: GENVIF_INPUT_VIF_XML_PATH=vif_alt2.xml
That is, users build with a given conf file which then points to a given vif xml, for example:
```
west build .... (uses prj.conf, and thereby vif.xml)
west build -DCONF_FILE=prj_alt2.conf .... (uses prj_alt2.conf, and thereby vif_alt2.xml)
```

Use case 2:
- prj.conf: GENVIF_INPUT_VIF_XML_PATH=vif.xml
- vif_alt1.xml
- vif_alt2.xml
Users are expected to build with prj.conf, but adjust vif xml using GENVIF_INPUT_VIF_XML_PATH, for example:
```
west build .... (uses prj.conf, and thereby vif.xml)
west build -DGENVIF_INPUT_VIF_XML_PATH=vif_alt2.xml .... (uses prj.conf, but with vif_alt2.xml instead)
```

Use case 3:
- prj.conf: GENVIF_INPUT_VIF_XML_PATH=vif.xml
- boards/foo.conf: GENVIF_INPUT_VIF_XML_PATH=boards/vif_foo.xml
Users are expected to build build for a board, and the board config fragment will point to alternative vif xml when required.
This means:
```
west build -DBOARD=<someboard> .... (uses prj.conf, and thereby vif.xml)
west build -DBOARD=foo .... (uses boards/foo.conf with prj.conf, and thereby boards/vif_foo.xml is used)
```

i'm not familiar with the vif and the intentions / under which circumstances the vif.xml would require changes for a given board.
How static is a vif.xml expected to be for a given application / application configuration ?
Is vif.xml expected to mostly be coupled to an application or to a board configuration.

Could you provide a couple of examples on when, how, why a vif.xml would change ?

Without such background knowledge it becomes very hard to review this PR and to provide best guidance on how to integrate this feature into the build system.

Reply:
Thanks for following up:
> For use case 1
> ...
> For use case 2, 3

I was not asking how it could be done, I was trying to understand what are the most likely / expected use-case that users will deal with.
You are free to pick one of the sketched use-cases, but I hoped you could describe the use-case intended, cause having an idea about most common use-case ( != all possible use-cases) then that give a better foundation for deciding correct approach.

> And I see in Zephyr based EC of ChromeOS, one board will be used by multiple projects, and USB C sub systems differs at project level.

Now this looks like what i'm asking for :+1: 
When you refer to `project1`, then i'm assuming that's a dedicated `prj.conf`, perhaps pointed to by `CONF_FILE=<file>` or using `APPLICATION_CONFIG_DIR=<dir>` which redirects the whole configuration setup to a dedicated folder, correct ?
So this is very similar to use case 1 in the above, would you agree ?

>  VIF is more specific to application or USBC subsystem it contains.

From this sentence and other comments, I have a slight feeling that it will be very hard to create a board default `vif.xml` that is suitable in most applications.
As the `vif.xml` is xml, and not like dts overlay files and config fragments, then it cannot as easily be extended.
I therefore start to wonder if a board default `vif.xml` is really that useful ?

Maybe application w/ board specific adjustments is all that really is needed, as was proposed here:
https://github.com/zephyrproject-rtos/zephyr/pull/52759#discussion_r1085022956
https://github.com/zephyrproject-rtos/zephyr/pull/52759#discussion_r1112951561
> > It may make more sense if it would lookup in <application>/boards/<board>_vif.xml or something like that.

but you actually did a different implementation than what was proposed.
You decided to use `BOARD_DIR` which is located in `<zephyr_base>/boards/<arch>/<board>`, which is not the same as `<application>/boards` that was proposed.

> it's user choice how they want to organise projects

correct, but having consistency in how Zephyr does things will make it easier to understand and provide a good starting point.
Hence the preference to keep an application specific file taking precedence over a board default from `BOARD_DIR`.
A board default in `<application>/boards` should take precendence over application default, and that is what was initially proposed.

Reply:
Thank you for the revert

`When you refer to project1, then i'm assuming that's a dedicated prj.conf, perhaps pointed to by CONF_FILE=<file> or using APPLICATION_CONFIG_DIR=<dir> which redirects the whole configuration setup to a dedicated folder, correct ?
So this is very similar to use case 1 in the above, would you agree ?`
yes, I agree

`As the vif.xml is xml, and not like dts overlay files and config fragments, then it cannot as easily be extended.
I therefore start to wonder if a board default vif.xml is really that useful ?`
If USB-C system is dedicated to board and at application level nothing related to UBS-C changes, then in that case it will be helpful

`but you actually did a different implementation than what was proposed.
You decided to use BOARD_DIR which is located in <zephyr_base>/boards/<arch>/<board>, which is not the same as <application>/boards that was proposed.`
As project structure seemed to be changing per user, I skipped looking for app/boards/board_vif.xml in cmake script as Kconfig option covers that case.